pkg/controller/watermarkpodautoscaler/replica_calculator.go

```diff
@@ -206,22 +207,26 @@ func getReplicaCount(logger logr.Logger, currentReplicas int32, wpa *v1alpha1.Wa
 	return replicaCount, utilizationQuantity.MilliValue()
 }
 
-func (c *ReplicaCalculator) getReadyPodsCount(namespace string, selector labels.Selector, readinessDelay time.Duration) (int32, error) {
-	podList, err := c.podLister.Pods(namespace).List(selector)
+func (c *ReplicaCalculator) getReadyPodsCount(target *autoscalingv1.Scale, selector labels.Selector, readinessDelay time.Duration) (int32, error) {
```

Comment:
should we rename this function `getAvailablePodsCount()`

Reply:
this returns the pods that are Ready though, not available

Reply:
should we use Available instead of ready?

Available means that the pods became ready at least once